torchmultimodal/models/video_gpt.py

```diff
@@ -0,0 +1,168 @@
+
+from torchmultimodal.modules.layers.attention import SelfAttention
+from torchmultimodal.modules.layers.position_embedding import (
+    BroadcastedPositionEmbedding,
+)
+
+
+def video_gpt(
+    input_shape: Tuple[int, int, int] = (16, 64, 64),
+    latent_shape: Tuple[int, int, int] = (8, 32, 32),
+    d_model: int = 576,  # https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/gpt.py#L177
+    n_head: int = 4,
+    dropout: float = 0.2,
+    attn_dropout: float = 0.3,
+    num_decoder_layers: int = 16,
+) -> MultimodalGPT:
+    """VideoGPT model
+
+    For the momemnt this is a prototype implementation which may not fully commit to the specifications
+    described in the paper `"VideoGPT: Video Generation using VQ-VAE and Transformers
+    "<https://arxiv.org/pdf/2104.10157.pdf>`_.
+    Source of parameters:
+        * Page 13 Table A.1 Column "BAIR / RoboNet / ViZDoom"
+        * Page 13 Table A.2 Column "BAIR / RoboNet"
+
+    Args:
+        input_shape (Tuple[int, int, int]): Shape of the input video data ``(time_seq_len, resolution, resolution)``.
+            Defaults to ``(16, 64, 64)``.
+        latent_shape (Tuple[int, int, int]): Shape of the encoded video data. This should be consistent with the actual
+            latent shape inferred by the video encoder, see :class:`~torchmultimodal.models.video_vqvae.VideoEncoder`.
+            Defaults to ``(8, 32, 32)``.
+        d_model (int): Dimension of the underlying transformer decoder.
+            See :py:class:`torchmultimodal.models.gpt.TransformerDecoderLayer`. Defaults to ``576``.
+        n_head (int): Number of attention heads used by the transformer decoder. Defaults to ``4``.
+        dropout (float): Dropout probability used by the projection layer of the transformer decoder.
+            Defaults to ``0.2``.
+        attn_dropout (float): Dropout probability used by the attention layer of the transformer decoder.
+            Defaults to ``0.3``.
+        num_decoder_layers (int): Number of transformer decoder layers. Defaults to ``16``.
+
+    Returns:
+        An instance of :class:`~torchmultimodal.models.gpt.MultimodalGPT`.
+    """
+    # constructs in and out tokenizers
+    in_tokenizer = video_vqvae()
+    out_tokenizer = video_vqvae()
+    num_in_tokens = in_tokenizer.num_embeddings  # codebook size
+    num_out_tokens = out_tokenizer.num_embeddings
+
+    # derived parameters
+    vqvae_latent_shape = in_tokenizer.latent_shape(input_shape)
+    if latent_shape != vqvae_latent_shape:
+        raise ValueError(
+            f"Latent shape required: {latent_shape} does not match that of VQVAE: {vqvae_latent_shape}"
+        )
+
+    # constructs projection layers
+    in_projection = nn.Linear(in_tokenizer.embedding_dim, d_model, bias=False)
+    in_projection.weight.data.normal_(std=0.02)
+    out_projection = nn.Linear(out_tokenizer.embedding_dim, d_model, bias=False)
+    out_projection.weight.data.normal_(std=0.02)
```

Comment:
Similar comment to the other PR: I'm not sure weight initialization should happen in the builder

Reply:
Updated in #283 . Will also update the callsite here.